spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,298 @@
+    dlg.hn.clear()
+    qtbot.keyClicks(dlg.hn, hn)
+    # fill out port
+    dlg.pn.clear()
+    qtbot.keyClicks(dlg.pn, str(pn))
+    # fill out username
+    dlg.un.clear()
+    qtbot.keyClicks(dlg.un, un)
+
+    # select ssh keyfile radio
+    dlg.kf_radio.setChecked(True)
+    assert dlg.kf.isEnabled()
+
+    # fill out ssh keyfile
+    dlg.kf.clear()
+    qtbot.keyClicks(dlg.kf, kf)
+
+    # fill out passphrase
+    dlg.kfp.clear()
+    qtbot.keyClicks(dlg.kfp, kfp)
+
+    # check save connection settings
+    dlg.save_layout.setChecked(False)
+
+    # Press ok and save connection settings
+    qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                     Qt.LeftButton)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get()
+    assert new_dlg.cf.text() == ""
+    assert not new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == ""
+    assert new_dlg.un.text() == ""
+    assert new_dlg.pn.text() == "22"
+    assert new_dlg.kf.text() == ""
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == ""
+
+
+def test_kernel_connection_dialog_remembers_input_password(
+    qtbot, connection_dialog_factory
+):
+    """Test to ensure that the dialog remembers user's kernel connection
+       settings and ssh password when the user checks the save checkbox"""
+
+    dlg = connection_dialog_factory.get()
+
+    cf_path = "test_cf_path"
+    un = "test_username"
+    hn = "test_hostname"
+    pn = 123
+    pw = "test_pw"
+
+    # fill out cf path
+    # dlg.cf.clear()
+    qtbot.keyClicks(dlg.cf, cf_path)
+    # check the remote kernel box
+    dlg.rm_group.setChecked(True)
+    # fill out hostname
+    # dlg.hn.clear()
+    qtbot.keyClicks(dlg.hn, hn)
+    # fill out port
+    dlg.pn.clear()
+    qtbot.keyClicks(dlg.pn, str(pn))
+    # fill out username
+    dlg.un.clear()
+    qtbot.keyClicks(dlg.un, un)
+
+    # select password radio
+    dlg.pw_radio.setChecked(True)
+    assert dlg.pw.isEnabled()
+
+    # fill out password
+    dlg.pw.clear()
+    qtbot.keyClicks(dlg.pw, pw)
+
+    # check save connection settings
+    dlg.save_layout.setChecked(True)
+
+    # Press ok and save connection settings
+    qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                     Qt.LeftButton)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get()
+    assert new_dlg.cf.text() == cf_path
+    assert new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == hn
+    assert new_dlg.un.text() == un
+    assert new_dlg.pn.text() == str(pn)
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.pw.text() == pw
+
+
+def test_kernel_connection_dialog_doesnt_remember_input_password(
+    qtbot, connection_dialog_factory
+):
+    """Test to ensure that the dialog doesn't remembers user's kernel
```

Comment:
"Ensure that the dialog" or "Test that the dialog"; "doesn't remember", and period at the end.